runners/spark/src/main/java/org/apache/beam/runners/spark/structuredstreaming/translation/helpers/EncoderHelpers.java

```diff
@@ -234,30 +194,33 @@ public Expression child() {
 
     @Override
     public ExprCode doGenCode(CodegenContext ctx, ExprCode ev) {
-      String accessCode =
-          ctx.addReferenceObj(
-              "beamCoderWrapper", beamCoderWrapper, BeamCoderWrapper.class.getName());
+      String accessCode = ctx.addReferenceObj("coder", coder, coder.getClass().getName());
       ExprCode input = child.genCode(ctx);
       String javaType = CodeGenerator.javaType(dataType());
+
       List<String> parts = new ArrayList<>();
       List<Object> args = new ArrayList<>();
       /*
             CODE GENERATED:
-            final $javaType ${ev.value} = ($javaType) $beamCoderWrapper.decode(${input.isNull}, ${input.value});
+            final ${javaType} ${ev.value} = (${input.isNull})
```

Comment:
This adds a little more generated code. Out of 4 runs, this PR seems to add from 0.1s to 0.2s to the queries. Maybe doing the ternary expression inside the compiled java code will be better.

Reply:
Ok I moved the ternary expression out of the generated code and now the execution numbers are better (in my machine).